### PR TITLE
west: hal_rpi_pico: Update Pico-SDK to 2.0.0

### PR DESCRIPTION
**Pull request by soburi**
Update Pico-SDK to 2.0.0 release.

- The directory structure has changed in 2.0.0, so we update it accordingly.
- Following the GPIO interface changes in pico-sdk 2.0.0.
- Some symbol names have been conflicted with introducing pico-sdk 2.0.0. Rename these.

### CODE REVIEW

**Comment by yonsch** [comment]
Ref https://github.com/zephyrproject-rtos/hal_rpi_pico/pull/6

**Review by yonsch — DISMISSED** [review]

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by ajf58** [comment]
FWIW I've been using this PR in my work with no issues. I think this is good-to-go/LGTM etc. Is there anything I can do to help progress this PR?

File: west.yml [hunk 1]
```diff
@@ -219,7 +219,7 @@ manifest:
         - hal
     - name: hal_rpi_pico
       path: modules/hal/rpi_pico
-      revision: fba7162cc7bee06d0149622bbcaac4e41062d368
+      revision: pull/6/head
```
Comment (fabiobaltieri) [inline west.yml:222]: `79ee0f9e058a6327fc943d2f2a19cf3ade107cec`

**Review by fabiobaltieri — APPROVED** [review]

**Comment by soburi** [comment]
@henrikbrixandersen 
Could you re-approve this please?

**Review by henrikbrixandersen — APPROVED** [review]

**Comment by soburi** [comment]
@yonsch 
I know you're busy, but could you please approve it?

**Review by dkalowsk — APPROVED** [review]

**Review by yonsch — APPROVED** [review]

**Pull request merged**